theano/tensor/opt.py

```diff
@@ -4957,7 +4971,7 @@ def local_reduce_join(node):
             node.inputs[0].owner and
             isinstance(node.inputs[0].owner.op, T.Join)):
         join = node.inputs[0].owner
-        if T.extract_constant(join.inputs[0]) != 0:
+        if T.extract_constant(join.inputs[0], only_process_constants=True) != 0:
```

Comment:
This one is also registered in register_uncanonicalize. Add this around line 5825 in this file:

register_uncanonicalize(topo_constant_folding, 'fast_compile', final_opt=True)